pkg/julia/manifest/parse.go

```diff
@@ -0,0 +1,118 @@
+package julia
+
+import (
+	"io"
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"github.com/aquasecurity/go-dep-parser/pkg/utils"
+
+	"golang.org/x/xerrors"
+)
+
+type Manifest struct {
+	JuliaVersion   string                  `toml:"julia_version"`
+	ManifestFormat string                  `toml:"manifest_format"`
+	Dependencies   map[string][]Dependency `toml:"deps"` // e.g. [[deps.Foo]]
+}
+type Dependency struct {
+	Dependencies []string `toml:"deps"` // by name. e.g. deps = ["Foo"]
+	UUID         string   `toml:"uuid"`
+	Version      string   `toml:"version"` // not specified for stdlib packages, which are of the Julia version
+}
+
+type Parser struct{}
+
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	var oldDeps map[string][]Dependency
+	var man Manifest
+	decoder := toml.NewDecoder(r)
+	// Try to read the old Manifest format. If that fails, try the new format.
+	if _, err := decoder.Decode(&oldDeps); err != nil {
+		r.Seek(0, io.SeekStart)
+		if _, err := decoder.Decode(&man); err != nil {
+			return nil, nil, xerrors.Errorf("decode error: %w", err)
+		}
+	}
+
+	// We can't know the Julia version on an old manifest.
+	// All newer manifests include a manifest version and a julia version.
+	if man.ManifestFormat == "" {
+		man = Manifest{
+			JuliaVersion:   "unknown",
+			ManifestFormat: "unknown",
+			Dependencies:   oldDeps,
+		}
+	}
```

Comment:
If i understand correctly - there are 2 manifest format for us:
- before 1.8
- after 1.8

But we don't use manifest format for parsing. So we can just compare with `nil` and remove `ManifestFormat`.
```suggestion
	if man.ManifestFormat == nil {
		man = Manifest{
			JuliaVersion:   "unknown",
			Dependencies:   oldDeps,
		}
	}
```

Correct me, if i have missed something.